utils/build.ps1

```diff
@@ -611,31 +682,27 @@ function Write-Summary {
 
   $TotalTime = [TimeSpan]::Zero
   foreach ($Entry in $TimingData) {
-    $TotalTime = $TotalTime.Add($Entry."Elapsed Time")
-  }
-
-  $SortedData = $TimingData | ForEach-Object {
-    $Percentage = [math]::Round(($_.("Elapsed Time").TotalSeconds / $TotalTime.TotalSeconds) * 100, 1)
-    $FormattedTime = "{0:hh\:mm\:ss\.ff}" -f $_."Elapsed Time"
-    [PSCustomObject]@{
-      "Build Step" = $_."Build Step"
-      Platform = $_.Platform
-      Arch = $_.Arch
-      "Elapsed Time" = $FormattedTime
-      "%" = "$Percentage%"
+    if (-not $Entry.Parent) {
+      $TotalTime = $TotalTime.Add($Entry."Elapsed Time")
     }
-  } | Sort-Object -Descending -Property "Elapsed Time"
+  }
+
+  $RootEntries = $TimingData | Where-Object { -not $_.Parent } | Sort-Object -Descending -Property "Elapsed Time"
+  $Result = foreach ($RootEntry in $RootEntries) {
+    Flatten-TimingEntry $RootEntry $TotalTime
+  }
 
   $FormattedTotalTime = "{0:hh\:mm\:ss\.ff}" -f $TotalTime
   $TotalRow = [PSCustomObject]@{
+    "Total Time" = $FormattedTotalTime
```

Comment:
Can we retain the original ordering? That feels more natural and much easier to find a particular timing.